jax/_src/pallas/primitives.py

```diff
@@ -170,41 +171,96 @@ def _atomic_abstract_eval(*avals_flat, args_tree, atomic_type: AtomicOpType):
   return _swap_abstract_eval(*avals_flat, args_tree=args_tree)
 
 
-atomic_rmw_p.def_effectful_abstract_eval(_atomic_abstract_eval)
-
-def atomic_rmw(x_ref_or_view, idx, val, *, mask: Any | None = None,
-               atomic_type: AtomicOpType):
+def _atomic_rmw(x_ref_or_view, idx, val, *, mask: Any | None = None,
+                atomic_type: AtomicOpType):
   x_ref, indexers = sp.get_ref_and_indexers(x_ref_or_view, idx, "atomic_rmw")
   args_flat, args_tree = tree_util.tree_flatten((x_ref, indexers, val, mask))
   return atomic_rmw_p.bind(
       *args_flat, args_tree=args_tree, atomic_type=atomic_type
   )
 
-atomic_xchg = functools.partial(atomic_rmw, atomic_type=AtomicOpType.XCHG)
-atomic_add = functools.partial(atomic_rmw, atomic_type=AtomicOpType.ADD)
-atomic_max = functools.partial(atomic_rmw, atomic_type=AtomicOpType.MAX)
-atomic_min = functools.partial(atomic_rmw, atomic_type=AtomicOpType.MIN)
-atomic_and = functools.partial(atomic_rmw, atomic_type=AtomicOpType.AND)
-atomic_or = functools.partial(atomic_rmw, atomic_type=AtomicOpType.OR)
-atomic_xor = functools.partial(atomic_rmw, atomic_type=AtomicOpType.XOR)
+def atomic_xchg(x_ref_or_view, idx, val, *, mask: Any | None = None):
+  """Atomically exchanges the given value with the value at the given index."""
+  return _atomic_rmw(
+      x_ref_or_view, idx, val, mask=mask, atomic_type=AtomicOpType.XCHG
+  )
+
+
+def atomic_add(x_ref_or_view, idx, val, *, mask: Any | None = None):
+  """Atomically adds a given value to the value at the given index."""
+  return _atomic_rmw(
+      x_ref_or_view, idx, val, mask=mask, atomic_type=AtomicOpType.ADD
+  )
+
+
+def atomic_max(x_ref_or_view, idx, val, *, mask: Any | None = None):
+  """Atomically computes the maximum of the given value and the value at the
```

Comment:
nit: not just computes, but I assume it stores the maximum in the ref?

same for `atomic_min` below.

Reply:
Good point. Updated the docs. PTAL.